Documentation/dotnet-primer.md

```diff
@@ -0,0 +1,147 @@
+
+- [Automatic memory management](garbage-collection.md)
+- Verifiable code - type and memory safety
+- Flexible native code compilation (enables JIT and AOT)
+- Fast native interop and access to native memory
+- Operating system agnostic 
+- Chip-agnostic (e.g. x86, ARM64) byte-code
+- Language-agnostic (e.g. C#, F#, VB) runtime
+- Flexible type system
+- Expressive (imperative and functional) programming languages
+- Programmable source code compiler (Roslyn)
+- Large standard class library
+- [.NET Class Libraries](class-libraries.md)
+- [Assembly format](assembly-format.md)
+- Package management
+
+TODO: Ensure that this list is correct. Write and link to a chapter for each of these defining features.
+
+.NET by Example
+===============
+
+.NET includes many features and capabilities that you will use on a daily basis for common tasks and needs. You can see these described below, with examples. Some of the same topics are discussed in more detail in documents linked above.
+
+Automatic Memory Management
+---------------------------
+
+[Garbage collection](garbage-collection.md) is the most well-known of .NET features. Developers do not need to actively manage memory, athough there are affordances to provide more information to the garbage collector (GC). C# includes the `new` keyword to allocate memory in terms of a particular type, and the `using` keyword to scope the usage of an object. The GC operates on a lazy approach to memory management, preferring application throughput to the immediate collection of memory.
+
+The following two lines both allocate. There is no analogous keyword to de-allocate the memory. The GC reclaims the memory sometime after the two variables go out of scope.
+
+	var title = ".NET Primer";
+	var list = new List<string>;
+
+Method variables normally go out of scope once a method completes, at which point they can be collected. The following example indicates to the GC that an object has gone out of scope, once the `using` statement has been completed, and the dependent `stream` object can be collected.
+
+	using(FileStream stream = GetFileStream(context))
+	{
+		//operations on the stream
+	}
+
+Memory Safety
+-------------
+
+One of the less obvious but quite far-reaching features that a garbage collector enables is memory safety.  The invariant of memory safety is very simple: a program is memory safe if it accesses only memory that has been allocated (and not freed). Dangling pointers are always bugs, and tracking them down is often quite difficult.
+
+The .NET runtime provides additional services, to complete the promise of memory safety, not naturally offered by a GC. It ensures that programs do not index off the end of an array or accessing a phantom field off the end of an object.
+
+The following example will throw as a result of memory safety.
+
+	int[] numbers = new int[42];
+	int number = numbers[42] // will throw (indexes are 0-based)
+
+Type Safety
+-----------
```

Comment:
I think this would make a better first section than "Automatic Memory Management", but that's just a personal opinion.